### PR TITLE
update submodules for llvm debug info and fancy assertions

### PR DESCRIPTION
**Pull request by glguy**

### CODE REVIEW

File: src/SAWScript/Crucible/JVM/Override.hs [hunk 1]
```diff
@@ -243,13 +243,15 @@ methodSpecHandler opts sc cc top_loc css h = do
                 case res of
                   Left (OF loc rsn)  ->
                     -- TODO, better pretty printing for reasons
-                    liftIO $ Crucible.abortExecBecause
-                      (Crucible.AssumedFalse (Crucible.AssumptionReason loc (show rsn)))
+                    liftIO
+                      $ Crucible.abortExecBecause
+                      $ Crucible.AssertionFailure
+                      $ Crucible.SimError loc
+                      $ Crucible.AssertFailureSimError "assumed false" (show rsn)
```
Comment (robdockins) [inline src/SAWScript/Crucible/JVM/Override.hs:250]: This isn't an assertion error, so we should just use the `InfeasibleBranch` reason.

Comment (robdockins) [reply src/SAWScript/Crucible/JVM/Override.hs:250]: Humm, sorry, looking into it a bit more, I guess this does make sense to consider as an assertion failure.

**Review by robdockins — CHANGES_REQUESTED** [review]
The changes here in SAW look good to me.

However, based on the CI results, it looks like we need to be a bit more careful about how we parse the metadata intrinsics.  I'm guessing clang has changed the signatures of those over time, so we'll need to figure out what the range of possibilities are.

**Comment by glguy** [comment]
> The changes here in SAW look good to me.
> 
> However, based on the CI results, it looks like we need to be a bit more careful about how we parse the metadata intrinsics. I'm guessing clang has changed the signatures of those over time, so we'll need to figure out what the range of possibilities are.

What range of LLVM versions do we need to support well?

I can imagine a couple of options:

- ignore outdated metadata
- update the test cases to use a more recent LLVM
- add a bunch of metadata patterns we can recognize.

**Comment by robdockins** [comment]
According to the SAW readme, we're aiming to support LLVM versions from 3.5 to 11.0.

**Comment by glguy** [comment]
> According to the SAW readme, we're aiming to support LLVM versions from 3.5 to 11.0.

I tried recompiling the salsa20 case with llvm 11 and saw couldn't handle it (independently of the metadata changes). I had to roll back to 9. I might have done something wrong but I don't think we actually have that complete of LLVM coverage.

LLVM 12 is out now, that'll need to be on the radar.

**Comment by robdockins** [comment]
We should definitely fix whatever is broken there with LLVM 11.

For the metadata stuff, I'd chose some reasonable version cutoff where the metadata looks more or less the way you need it to (they've changed their format quite a bit) and make sure to handle all formats from there forward, and gracefully ignore older ones.

**Review by robdockins — APPROVED** [review]

**Pull request merged**